test/e2e/stack_test.go

```diff
@@ -29,65 +29,107 @@ import (
 		WithKibanaRef(kbRef).
 		WithRestrictedSecurityContext()
-	apmUpdated := apm.WithVersion(updatedVersion)
 	ent := enterprisesearch.NewBuilder("ent").
 		WithNodeCount(1).
 		WithVersion(initialVersion). // pre 8.x doesn't require any config, but we change the version after calling
 		WithoutConfig().             // NewBuilder which relies on the version from test.Ctx(), so removing config here
 		WithElasticsearchRef(esRef).
 		WithRestrictedSecurityContext()
-	entUpdated := ent.WithVersion(updatedVersion)
-	logstash := logstash.NewBuilder("ls").
-		WithVersion(initialVersion) // pre 8.x doesn't require any config, but we change the version after calling
-	logstashUpdated := logstash.WithVersion(updatedVersion)
 	fb := beat.NewBuilder("fb").
 		WithType(filebeat.Type).
 		WithRoles(beat.AutodiscoverClusterRoleName).
 		WithVersion(initialVersion).
 		WithElasticsearchRef(esRef).
 		WithKibanaRef(kbRef)
 	fb = beat.ApplyYamls(t, fb, beattests.E2EFilebeatConfig, beattests.E2EFilebeatPodTemplate)
-	fbUpdated := fb.WithVersion(updatedVersion)
 
-	initialBuilders := []test.Builder{es, kb, apm, ent, fb, logstash}
-	updatedBuilders := []test.Builder{esUpdated, kbUpdated, apmUpdated, entUpdated, fbUpdated, logstashUpdated}
+	esUpdated := es.WithVersion(test.LatestReleasedVersion8x)
+	kbUpdated := kb.WithVersion(test.LatestReleasedVersion8x)
+	apmUpdated := apm.WithVersion(test.LatestReleasedVersion8x)
+	entUpdated := ent.WithVersion(test.LatestReleasedVersion8x)
+	fbUpdated := fb.WithVersion(test.LatestReleasedVersion8x)
+
+	initialBuilders := []test.Builder{es, kb, apm, ent, fb}
+	updatedBuilders := []test.Builder{esUpdated, kbUpdated, apmUpdated, entUpdated, fbUpdated}
+
+	stackVersions := StackResourceVersions{
+		Elasticsearch:    ref(k8s.ExtractNamespacedName(&es.Elasticsearch)),
+		Kibana:           ref(k8s.ExtractNamespacedName(&kb.Kibana)),
+		ApmServer:        ref(k8s.ExtractNamespacedName(&apm.ApmServer)),
+		EnterpriseSearch: ref(k8s.ExtractNamespacedName(&ent.EnterpriseSearch)),
+		Beat:             ref(k8s.ExtractNamespacedName(&fb.Beat)),
+	}
+
+	return initialBuilders, updatedBuilders, stackVersions
+}
+
+func TestVersionUpgradeOrdering(t *testing.T) {
+	initialBuilders, updatedBuilders, stackVersions := initialBuildersToUpgrade(t, test.LatestReleasedVersion7x)
+	runVersionUpgradeOrdering(t, initialBuilders, updatedBuilders, stackVersions)
+}
+
+func TestVersionUpgradeOrderingWithLogstash(t *testing.T) {
+	initialVersion := "8.6.0"
+	initialBuilders, updatedBuilders, stackVersions := initialBuildersToUpgrade(t, "8.6.0")
+
+	ls := logstash.NewBuilder("ls").WithVersion(initialVersion).
+		WithElasticsearchRefs(
+			// associate logstash to the es ref stored in the stackVersions
+			logstashv1alpha1.ElasticsearchCluster{
+				ObjectSelector: commonv1.ObjectSelector{
+					Namespace: stackVersions.Elasticsearch.ref.Namespace,
+					Name:      stackVersions.Elasticsearch.ref.Name,
+				},
+				ClusterName: "es",
```

Comment:
@pebrc Would you say more than that?

Reply:
That's even better than what I had in mind. I thought we just document that we don't have an association. But you are right why not just create the association.